lib/record_store/provider.rb

```diff
@@ -152,24 +153,22 @@ def retry_on_connection_errors(
         )
 
         loop do
-          begin
-            return yield
-          rescue UnparseableBodyError
-            raise if max_retries <= 0
-            max_retries -= 1
-
-            waiter.wait(message: 'Waiting to retry after receiving an unparseable response')
-          rescue Net::OpenTimeout, Errno::ETIMEDOUT
-            raise if max_timeouts <= 0
-            max_timeouts -= 1
-
-            $stderr.puts('Retrying after a connection timeout')
-          rescue Errno::ECONNRESET
-            raise if max_conn_resets <= 0
-            max_conn_resets -= 1
-
-            waiter.wait
-          end
+          return yield
```

Comment:
Rubocop detected redundant `begin` statements, and that appears to be consistent with the [Ruby Style Guide](https://rubystyle.guide/#no-BEGIN-blocks)

Reply:
This is a good cop to have on. :+1: